### ISSUE TITLE
Don't leave empty field tables in the database

### ISSUE THREAD

**Issue by mikemccaffrey** [backdrop-contrib/multifield#20]
We should probably figure out a way to get this to work without doing any "unholy things" with entities and fields.

One thing I've discovered is that field tables are being created in the database when you add a subfield, but then are not actually being used to store the values, which are put in columns in the table for the multifield. Do we perhaps want things to function more like Field Collection, where an entity type is created for the parent field, and then its values are stored in regular field tables?

What other unholy things is this module doing?

**Comment by laryn** [comment]
@mikemccaffrey The "unholy" language was just copied from the Drupal module but I'm all for making it work without "unholy things".

Regarding the empty field tables, is that the same as this one?
https://github.com/backdrop-contrib/multifield/issues/15

Are there any cons to doing it that way? e.g. this is also from the Drupal module:

> As much as I love Field collection, it still has to save actual entities, and can cause performance problems due to having to load all the referenced field collection entities on node, or parent entity load.

**Comment by mikemccaffrey** [comment]
> Regarding the empty field tables, is that the same as this one? #15

Nope, these are individual tables create for each field, probably by the core field api, even though the multifield then doesn't use the tables to store the data.

**Comment by jenlampton** [comment]
> Do we perhaps want things to function more like Field Collection

No. 

The whole point of multifield was to make a more performant alternative to field collection. Some of us need both collections of fields, AND fast sites :) Keeping all the data in the same database table (instead of spread across multiple field tables) makes queries faster. 

The only reason we are using the field API at all, is so that the field UI can be used to assemble a multifield, and so that everyone would have the same choices of sub-fields as they would with something like field collection.

The data storage, however, needs to remain normalized to keep it fast, so, rather un-field-like .

People who want to use Entity and Field API the way they were intended, should choose to use Paragraphs instead of this module. 

> field tables are being created in the database when you add a subfield, but then are not actually being used to store the values

Yuck. :( This should be fixed. Re-titling this issue so we can work on this problem here.
